src/Microsoft.AspNet.Mvc.Core/ActionResults/FileResult.cs

```diff
@@ -24,15 +24,25 @@ public abstract class FileResult : ActionResult
         /// the provided <paramref name="contentType"/>.
         /// </summary>
         /// <param name="contentType">The Content-Type header of the response.</param>
-        protected FileResult(string contentType)
+        protected FileResult([NotNull] string contentType)
+            : this(new MediaTypeHeaderValue(contentType))
+        {
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="FileResult"/> instance with
+        /// the provided <paramref name="Microsoft.Net.Http.Headers.MediaTypeHeaderValue"/>.
+        /// </summary>
+        /// <param name="contentType">The Content-Type header of the response.</param>
+        protected FileResult([NotNull] MediaTypeHeaderValue contentType)
         {
             ContentType = contentType;
         }
 
         /// <summary>
-        /// Gets or sets the Content-Type header value that will be written to the response.
+        /// Gets the <see cref="MediaTypeHeaderValue"/> representing the Content-Type header of the response.
         /// </summary>
-        public string ContentType { get; set; }
+        public MediaTypeHeaderValue ContentType { get; }
```

Comment:
Yeah I think it's intentional because the underlying property on the `MediaTypeHeaderValue` is get-only. It's passed into the ctor of that type and is read-only. So this is a minor breaking change from MVC 5 (or Web API 2 or something), but we discussed on Friday and it's fine.